spyder/plugins/explorer/tests/test_explorer.py

```diff
@@ -32,6 +40,48 @@ def project_explorer(qtbot):
     return widget
 
 
+@pytest.fixture(params=[['script.py'],
+                        ['script.py', 'script1.py', 'testdir/script2.py'],
+                        ['subdir/innerdir/text.txt', 'testdir']])
+def create_folders_files(tmpdir, request):
+    """A project directory with dirs and files for testing."""
+    project_dir = to_text_string(tmpdir.mkdir('project'))
+    destination_dir = to_text_string(tmpdir.mkdir('destination'))
+    list_paths = []
+    for item in request.param:
+        if osp.splitext(item)[1]:
+            if osp.split(item)[0]:
+                dirs, fname = osp.split(item)
+                dirpath = osp.join(project_dir, dirs)
+                if not osp.exists(dirpath):
+                    os.makedirs(dirpath)
+                    item_path = osp.join(dirpath, fname)
+            else:
+                item_path = osp.join(project_dir, item)
+        else:
+            dirpath = osp.join(project_dir, item)
+            if not osp.exists(dirpath):
+                os.makedirs(dirpath)
+                item_path = dirpath
+        if not osp.isdir(item_path):
+            with open(item_path, 'w') as fh:
+                fh.write("File Path:\n" + str(item_path) + '\n')
```

Comment:
Do the ``.replace(os.sep, '/')`` here so you don't have to do it every time in your test functions and you avoid any mess with string escapes.